src/main/java/fr/theorozier/webstreamer/display/render/DisplayLayerManager.java

```diff
@@ -51,7 +54,11 @@ public boolean cleanup(long now) {
 
     @Override
     @NotNull
-    protected URI getLayerKey(Key key) {
+    protected Object getLayerKey(Key key) {
+        String path = key.uri().getPath();
+        if (path.endsWith(".svg")) {
+            return svgMap.getLayerKey(key); // we need to call this somewhere.
+        }
```

Comment:
This modification is not needed, because the main key is URI, the map that you return in `getNewLayer` already handles the size-dependant dispatching, this is recursive.

Reply:
Resolved in https://github.com/mindstorm38/fabric-webstreamer/pull/28/commits/1c0c2e9fdf6e0900dbf82b5f743fdc32ea4a4e03